pallet/src/lib.rs

```diff
@@ -98,6 +103,22 @@ pub mod pallet {
     #[pallet::getter(fn authorities)]
     pub(super) type Authorities<T: Config> = StorageValue<_, Vec<T::AuthorityId>, ValueQuery>;
 
+    // SBP M1 review: IMHO you could probably simplify some of the code related
+    // to the SessionPeriod, MillisecsPerBlock & UnitCreationDelay values
+    // (as they don't seem to be likely to change after the initial value was set).
+    //
+    // That is, by defining 3 types in the pallet's configuration trait, and apply
+    // the #[pallet:constant] attribute e.g. with SessionPeriod
+    //
+    // #[pallet::config]
+    // pub trait Config: frame_system::Config + pallet_session::Config {
+    //     // ...
+    //     #[pallet::constant]
+	//     type SessionPeriod: Get<u32>;
+    // }
+
+    // SBP M1 review: based on the comment above, you could probably remove the following code
+    // vvvvvvv
```

Comment:
This indeed simplifies the logic, but we opted for the more involved one so that we may easily change those parameters in CI and for experiments.